automation/patch_igz/patch_remote.py

```
@@ -40,101 +40,120 @@ class Consts:
         api_container = "mlrun-api"
         log_collector_container = "mlrun-log-collector"
 
-    def __init__(self, conf_file, patch_file, reset_db, tag, log_collector):
+    def __init__(
+        self,
+        conf_file: str,
+        patch_file: str,
+        reset_db: str,
+        image_tag: str,
+        patch_log_collector_image: bool,
+        patch_mlrun_image: bool,
+        patch_api: bool,
+    ):
         self._config = yaml.safe_load(conf_file)
         patch_yaml_data = yaml.safe_load(patch_file)
         self._deploy_patch = json.dumps(patch_yaml_data)
         self._reset_db = reset_db
-        self._tag = tag
-        self._patch_log_collector = bool(log_collector)
+        self._image_tag = image_tag
+        self._patch_log_collector_image = bool(patch_log_collector_image)
         self._validate_config()
+        self._patch_mlrun_image = patch_mlrun_image
+        self._patch_api = patch_api
+
+        self._make_target_to_image_name = {
+            "api": "mlrun-api",
+            "mlrun": "mlrun",
+            "log-collector": "log-collector",
+        }
 
         nodes = self._config["DATA_NODES"]
         if not isinstance(nodes, list):
             nodes = [nodes]
         self._nodes = nodes
 
-    def patch_mlrun_api(self):
+    def patch(self):
+        version = self._get_current_version()
+        image_tag = self._get_image_tag(version)
+        targets = []
+        if self._patch_api:
+            targets.append("api")
+        if self._patch_mlrun_image:
+            targets.append("mlrun")
+        if self._patch_log_collector_image:
+            targets.append("log-collector")
+        if not targets:
+            raise ValueError("No targets to patch")
+
+        target_to_built_images = self._make_targets(
+            targets=targets,
+            image_tag=image_tag,
+        )
+        # Docker login and push images
         self._docker_login_if_configured()
```

move this before making the targets to fail fast if needed